templates/definition/charger/heishamon.yaml

```
@@ -0,0 +1,178 @@
+    required: true
+    default: 60
+    type: int
+  - name: dhw_boost_temperature
+    description:
+      de: Warmwassertemperatur bei Boost
+      en: DHW temperature when boosted
+    unit: °C
+    help:
+      de: min. 40°C / max. 75°C
+      en: min. 40°C / max. 75°C
+    required: true
+    default: 65
+    type: int
+  - name: dhw_dimm_temperature
+    description:
+      de: Warmwassertemperatur bei Dimmung
+      en: DHW temperature when dimmed
+    unit: °C
+    help:
+      de: min. 40°C / max. 75°C
+      en: min. 40°C / max. 75°C
+    required: true
+    default: 40
+    type: int
+  - name: power_source
+    description:
+      de: Datenquelle für die aktuelle Leistungsaufnahme
+      en: Datasource for the actual power consumption
+    help:
+      de: s0_sum = Summe der zwei S0 Zähler, s0_1 = S0 Zähler 1, s0_2 = S0 Zähler 2, hp = Interner Zähler der Wärmepumpe
+      en: s0_sum = sum of both S0 meters, s0_1 = S0 meter 1, s0_2 = S0 meter 2, hp = internal heatpump meter
+    required: false
+    default: s0_sum
+    type: string
+render: |
+  type: sgready
+  setmode:
+    source: switch
+    switch:
+      - case: 1 # dimm
+        set:
+          source: mqtt
+          {{- if eq .heating_zone "dhw" }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetDHWTemp
+          payload: {{.dhw_dimm_temperature}}
+          {{- else }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetZ{{.heating_zone}}HeatRequestTemperature
+          payload: {{.dimm_temperature_offset}}
+          {{- end }}
+      - case: 2 # normal
+        set:
+          source: mqtt
+          {{- if eq .heating_zone "dhw" }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetDHWTemp
+          payload: {{.dhw_normal_temperature}}
+          {{- else }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetZ{{.heating_zone}}HeatRequestTemperature
+          payload: 0
+          {{- end }}
+      - case: 3 # boost
+        set:
+          source: mqtt
+          {{- if eq .heating_zone "dhw" }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetDHWTemp
+          payload: {{.dhw_boost_temperature}}
+          {{- else }}
+          topic: {{.heishamon_mqtt_topic}}/commands/SetZ{{.heating_zone}}HeatRequestTemperature
+          payload: {{.boost_temperature_offset}}
+          {{- end }}
+  getmode:
+    {{- if eq .heating_zone "dhw" }}
+    source: mqtt
+    topic: {{.heishamon_mqtt_topic}}/main/DHW_Target_Temp
+    jq: 'if . == {{.dhw_dimm_temperature}} then 1 elif . == {{.dhw_boost_temperature}} then 3 else 2 end'
+    {{- else }}
+    source: mqtt
+    topic: {{.heishamon_mqtt_topic}}/main/Z{{.heating_zone}}_Heat_Request_Temp
+    jq: 'if . < 0 then 1 elif . > 0 then 3 else 2 end'
+    {{- end }}
+  temp:
+    {{- if eq .heating_zone "dhw" }}
+    source: mqtt
+    topic: {{.heishamon_mqtt_topic}}/main/DHW_Temp
+    {{- else }}
+    source: mqtt
+    topic: {{.heishamon_mqtt_topic}}/main/Main_Outlet_Temp
+    {{- end }}
```

```suggestion
    source: mqtt
    topic: {{.heishamon_mqtt_topic}}/main/{{ if eq .heating_zone "dhw" }}DHW{{ else }}Main_Outlet{{ end }}_Temp
```

Simplify all of these